activerecord/test/cases/tasks/postgresql_rake_test.rb

```diff
@@ -139,45 +181,70 @@ def drop_database(*); end
-        ActiveRecord::Base.expects(:clear_active_connections!)
+        with_stubbed_connection_clear_active_connections do
+          ActiveRecord::Base.stub(:establish_connection, true) do
+            ActiveRecord::Base.expects(:clear_active_connections!)
 
-        ActiveRecord::Tasks::DatabaseTasks.purge @configuration
+            ActiveRecord::Tasks::DatabaseTasks.purge @configuration
+          end
+        end
       end
 
       def test_establishes_connection_to_postgresql_database
-        ActiveRecord::Base.expects(:establish_connection).with(
-          "adapter"            => "postgresql",
-          "database"           => "postgres",
-          "schema_search_path" => "public"
-        )
-
-        ActiveRecord::Tasks::DatabaseTasks.purge @configuration
+        ActiveRecord::Base.stubs(:establish_connection).returns(true)
+        with_stubbed_connection_clear_active_connections do
+          ActiveRecord::Base.expects(:establish_connection).with(
+            "adapter"            => "postgresql",
+            "database"           => "postgres",
+            "schema_search_path" => "public"
+          )
+
+          ActiveRecord::Tasks::DatabaseTasks.purge @configuration
+        end
       end
 
       def test_drops_database
-        @connection.expects(:drop_database).with("my-app-db")
+        with_stubbed_connection_clear_active_connections do
+          ActiveRecord::Base.stub(:establish_connection, true) do
+            @connection.expects(:drop_database).with("my-app-db")
 
-        ActiveRecord::Tasks::DatabaseTasks.purge @configuration
+            ActiveRecord::Tasks::DatabaseTasks.purge @configuration
+          end
+        end
       end
 
       def test_creates_database
-        @connection.expects(:create_database).
-          with("my-app-db", @configuration.merge("encoding" => "utf8"))
+        with_stubbed_connection_clear_active_connections do
+          ActiveRecord::Base.stub(:establish_connection, true) do
+            @connection.expects(:create_database).
+              with("my-app-db", @configuration.merge("encoding" => "utf8"))
 
-        ActiveRecord::Tasks::DatabaseTasks.purge @configuration
+            ActiveRecord::Tasks::DatabaseTasks.purge @configuration
+          end
+        end
       end
 
       def test_establishes_connection
-        ActiveRecord::Base.expects(:establish_connection).with(@configuration)
+        ActiveRecord::Base.stubs(:establish_connection).returns(true)
+        with_stubbed_connection_clear_active_connections do
+          ActiveRecord::Base.expects(:establish_connection).with(@configuration)
 
-        ActiveRecord::Tasks::DatabaseTasks.purge @configuration
+          ActiveRecord::Tasks::DatabaseTasks.purge @configuration
+        end
       end
+
+      private
+
+        def with_stubbed_connection_clear_active_connections
+          ActiveRecord::Base.stub(:connection, @connection) do
+            ActiveRecord::Base.stub(:clear_active_connections!, true) do
```

Comment:
why do we need to stub `clear_active_connections!`?

Reply:
No we don't! Removing.